pybossa/view/bloomberg.py

```diff
@@ -70,11 +71,30 @@ def handle_bloomberg_response():
         current_app.logger.error('BSSO auth error(s): %s %s', errors, error_reason)
         flash(gettext('There was a problem during the sign in process.'), 'error')
         return redirect(url_for('home.home'))
-    if auth.is_authenticated:
+    elif auth.is_authenticated:
         attributes = auth.get_attributes()
         user = user_repo.get_by(email_addr=unicode(attributes['emailAddress'][0]).lower())
         return _sign_in_user(user, next_url=request.form.get('RelayState'))
-    else:
+    elif not auth.is_authenticated:
         current_app.logger.error('BSSO authentication failed')
         flash(gettext('Authentication failed.'), 'error')
         return redirect(url_for('home.home'))
+    else:
+        attributes = auth.get_attributes()
+        user_data = {}
+        try:
+            user_data['fullname']   = attributes['FirstName'][0] + " " + attributes['LastName'][0]
+            user_data['email_addr'] = attributes['emailAddress'][0]
+            user_data['name']       = attributes['LoginID'][0]
+            user_data['password']   = u"pAssw0rd"
+            create_account(user_data)
+            user = user_repo.get_by(email_addr=unicode(attributes['emailAddress'][0]).lower())
+            return _sign_in_user(user, next_url=request.form.get('RelayState'))
+        except DBIntegrityError as dberror:
+            print(dberror)
+            flash(gettext('There was a problem logging into your account. Please contact a Gigwork administrator'), 'error')
+            return redirect(url_for('home.home'))
+        except Exception as ex:
+            print(ex)
```

Comment:
Should we consider using `logger.error` or `logger.info` etc here, rather than print? Not sure of the convention in this file. Thanks.

Reply:
Yes that's a much better idea. Thank you!

Reply:
`logger.exception` is the way to go in these cases - it will output the stack trace and the exception information.